python/pyspark/ml/tests.py

```diff
@@ -230,6 +230,15 @@ def test_pipeline(self):
         self.assertEqual(5, transformer3.dataset_index)
         self.assertEqual(6, dataset.index)
 
+    def test_identity_pipeline(self):
+        dataset = MockDataset()
+
+        def doTransform(pipeline):
+            pipeline_model = pipeline.fit(dataset)
+            return pipeline_model.transform(dataset)
+        self.assertEqual(dataset.index, doTransform(Pipeline()).index)
+        self.assertEqual(dataset.index, doTransform(Pipeline(stages=[])).index)
```

Comment:
Maybe, but these are no different than any other param and we usually don't test that we can get/set each one.  I think it's safe to assume that `setParams` will set the given value and `getStages` will return the value set or default.